### PR TITLE
Add PineconeVectorStore

### PR DESCRIPTION
**Pull request by mtutty**
Adds a vector store class for Pinecone along with sample app in apps/simple/pinecone-vector-store

### CODE REVIEW

File: apps/simple/pinecone-vector-store/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Pinecone Vector Store
```
Comment (marcusschiesser) [inline apps/simple/pinecone-vector-store/README.md:1]: please move example to `/examples/` and try with the local `llamaindex` build (see https://github.com/run-llama/LlamaIndexTS/blob/main/examples/README.md#using-the-local-build)

File: turbo.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "$schema": "https://turbo.build/schema.json",
+  "globalEnv": ["PINECONE_CHUNK_SIZE", "PINECONE_INDEX_NAME"],
```
Comment (marcusschiesser) [inline turbo.json:3]: we added the new env variables to packages/eslint-config-custom/index.js

**Comment by mtutty** [comment]
Hi @marcusschiesser I think I resolved all of the issues you noted.  Also re-ran the query.ts example for Pinecone to make sure everything was good end-to-end.  Let me know what else you find.

File: examples/pinecone-vector-store/load-docs.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
+// load-docs.ts
+import fs from "fs/promises";
+import {
+  SimpleDirectoryReader,
+  storageContextFromDefaults,
+  VectorStoreIndex,
+} from "llamaindex";
+import { PineconeVectorStore } from "../../../packages/core/src/storage/vectorStore/PineconeVectorStore";
```
Comment (marcusschiesser) [inline examples/pinecone-vector-store/load-docs.ts:8]: please also import this from "llamaindex" (works if you use the local build containing your class - see README.md)

**Review by marcusschiesser — CHANGES_REQUESTED** [review]
please update import (as in comment)

File: examples/pinecone-vector-store/load-docs.ts (hunk 3 above)
Comment (mtutty) [reply examples/pinecone-vector-store/load-docs.ts]: Done!

**Comment by mtutty** [comment]
@marcusschiesser Should be ready to check again.

**Comment by marcusschiesser** [comment]
> @marcusschiesser Should be ready to check again. I noticed an upstream conflict in pnpm-lock.yaml but it wouldn't resolve for me, even when I deleted and re-created the file with pnpm i. Let me know if there's something I should do there.

@mtutty hmm, I guess the `main` branch in your fork is outdated:
image
You can sync your fork, to prevent that from happening, see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

Otherwise, your PR looks great!

Btw, we now have VectorDB support in `create-llama`, it would make sense to add Pinecone to `create-llama` in this PR too. Do you want to have a look into that, too?

It's just three steps:
1. Adding a new `pinecone` folder to https://github.com/run-llama/LlamaIndexTS/tree/main/packages/create-llama/templates/components/vectordbs - the package should support the factory method `createChatEngine(llm: LLM)`, similar to the MongoDB implementation
2. Adding `pinecone` to this type:
https://github.com/run-llama/LlamaIndexTS/blob/e55d41f5df3c365c45f3b5e2d98cd679ab7a8972/packages/create-llama/templates/types.ts#L7
3. Add `pinecone` as a new option:
https://github.com/run-llama/LlamaIndexTS/blob/e55d41f5df3c365c45f3b5e2d98cd679ab7a8972/packages/create-llama/questions.ts#L255-L261

I am happy to help with that!

To test `create-llama`, you have to call:
```
cd packages/create-llama
pnpm run build
./packages/create-llama/dist/index.js
```

**Comment by mtutty** [comment]
HI @marcusschiesser Yeah, I tried getting my work branch up to date with your main and found that conflict.  I'll go back and start with my main as you suggested.  Might come back to you for guidance on how to resolve the conflict (which I expect will still be there).  It wasn't a logic or code problem, it had to do with the tooling and workflow of the project.  I'm traveling today, will get back to it Wed or Thursday and let you know what I find.
In regards to the create_llama thing, I can start a new PR after this one's cleared!

**Comment by mtutty** [comment]
Hi @marcusschiesser I was able to sync my main to run-llama/main, and then get my work branch up to date with that.  I had to run `pnpm i --no-frozen-lockfile` to resolve the pnpm-lock.yaml conflict, and that succeeded.  But when I went to commit the resolved changes, the local commit hook gave this failure:

```
llamaindex:lint: /workspaces/LlamaIndexTS/packages/core/dist/index.js
llamaindex:lint:   1686:112  error  ASTRA_DB_ID is not listed as a dependency in turbo.json         turbo/no-undeclared-env-vars
llamaindex:lint:   1687:116  error  ASTRA_DB_REGION is not listed as a dependency in turbo.json     turbo/no-undeclared-env-vars
llamaindex:lint:   1688:120  error  ASTRA_DB_NAMESPACE is not listed as a dependency in turbo.json  turbo/no-undeclared-env-vars

```
Not sure this is my problem to fix, or how this got into run-llama:main.  Let me know how to proceed.

**Comment by marcusschiesser** [comment]
@mtutty The issue was that your branch was not rebased on the `main` branch from this repo. I fixed that and merged your changes in https://github.com/run-llama/LlamaIndexTS/pull/296 - I saw that you merged `main` into your branch - for feature branches, I would recommend rebasing your branch onto `main` - then you'll end up having a linear commit history which is easier to deal with.
Can you please test your example with the current `main` branch again? Then you can close this PR.
Thanks a lot for the contribution! Let me know if you have any questions about adding pinecone to `create-llama`

**Pull request closed without merging**